contracts/src/StakeTable.sol

```diff
@@ -101,27 +111,48 @@ contract StakeTable is AbstractStakeTable {
         return nodes[_hashBlsKey(blsVK)];
     }
 
+    /// @dev Helper function to avoid code duplication. Contains the core logic of a queue for
+    /// finding the next epoch.
+    /// @param queueType can be Registration or Exit.
+    /// @return Number of the epoch when a user can register/exit.
+    function _nextEpoch(QueueType queueType) private returns (uint64) {
```

Comment:
1. remove `QueueType` abstraction, simply flatten them
2. specialize the logic into the original `nextRegisration/ExitEpoch` function

while I understand the intent is for code dedup, I still find these structs introducing unnecessary overheads.
since the logic is simple, won't be lots of duplications and IMO the code will clearer

Reply:
Ok, done in bc88335. Note that the `QueueType` enum has been moved to the tests. I personally prefer avoiding code duplication in the test as there is a risk to introduce error if the logic of the queue changes.